provision/internal/operator/native/gardener/gardener.go

```diff
@@ -0,0 +1,350 @@
+package gardener
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"time"
+
+	gardenerTypes "github.com/gardener/gardener/pkg/apis/core/v1beta1"
+	gardenerApi "github.com/gardener/gardener/pkg/client/core/clientset/versioned/typed/core/v1beta1"
+	"github.com/kyma-incubator/hydroform/provision/internal/operator/native/gardener/aws"
+	"github.com/kyma-incubator/hydroform/provision/internal/operator/native/gardener/azure"
+	"github.com/kyma-incubator/hydroform/provision/internal/operator/native/gardener/gcp"
+	"github.com/kyma-incubator/hydroform/provision/types"
+	"github.com/pkg/errors"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+/*-- Gardener native operator --*/
+
+func Create(ops *types.Options, cfg map[string]interface{}) (*types.ClusterInfo, error) {
+	client, err := seedClient(cfg["credentials_file_path"].(string))
+	if err != nil {
+		return nil, errors.Wrap(err, "error creating the gardener client from credentials")
+	}
+	shoot, err := toShoot(cfg)
+	if err != nil {
+		return nil, errors.Wrap(err, "error generating shoot spec from config")
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Minute) // TODO use the timeouts in the Options param
```

Comment:
Is this comment still valid? Or do you want to plan this feature for future work

Reply:
Yes it is, I added a hardcoded timeout for now but this would need to be properly wired into the options and also be passed on by the client using the API (the CLI).
I will create a follow up for this.